script.js

```diff
@@ -0,0 +1,71 @@
+let defaultResult = 0;
+
+let currentResult = defaultResult;
+
+let logEntries = [];
+
+function getUserInput() {
+  return parseInt(userInput.value);
+}
+
+// the description & result output of calculation
+
+function createOutput(numberBeforeCalc, operator, calcNumber) {
+  const calcDescription = `${numberBeforeCalc} ${operator} ${calcNumber}`;
+  outputResult(currentResult, calcDescription);
+}
+
+// object inside array
+
+function writeToLog(
+  operationIdentifier,
+  prevResult,
+  numberEnteredByUser,
+  finalResult
+) {
+  const logEntry = {
+    operation: operationIdentifier,
+    PrevResult: prevResult,
+    number: numberEnteredByUser,
+    output: finalResult,
+  };
+
+  logEntries.push(logEntry);
+  console.log(logEntries);
+}
+
+function addition() {
+  const initialResult = currentResult;
+  currentResult = currentResult + getUserInput();
+  createOutput(initialResult, "+", getUserInput());
+  writeToLog("ADD", initialResult, getUserInput(), currentResult);
+}
+
+function subtraction() {
+  const initialResult = currentResult;
+  currentResult = currentResult - getUserInput();
+  createOutput(initialResult, "-", getUserInput());
+  writeToLog("SUBT", initialResult, getUserInput(), currentResult);
+}
+
+function multiplication() {
+  const initialResult = currentResult;
+  currentResult = currentResult * getUserInput();
+  createOutput(initialResult, "*", getUserInput());
+  writeToLog("MULTIPLY", initialResult, getUserInput(), currentResult);
+}
+
+function division() {
+  const initialResult = currentResult;
+  currentResult = currentResult / getUserInput();
+  createOutput(initialResult, "/", getUserInput());
+  writeToLog("DIVIDE", initialResult, getUserInput(), currentResult);
+}
+
+addBtn.addEventListener("click", addition);
```

Comment:
addBtn.addEventListener("click", addition());  
what will happen if i write like this @Rehan-Khan-Engineer

Reply:
I think if you add parenthesis it means you are calling the function right then and there! 
A function name without parenthesis represents you are only registering  it and when the click occurs then only it gets executed. @anask601

Reply:
Great @Rehan-Khan-Engineer